tests/50federation/33room-get-missing-events.pl

```diff
@@ -413,3 +411,125 @@ sub sytest_user_and_room_fixture {
          Future->done;
       });
    };
+
+test "Outbound federation will ignore a missing event with bad JSON for room version 6",
+   requires => [ $main::OUTBOUND_CLIENT, $main::INBOUND_SERVER,
+                 local_user_and_room_fixtures(
+                    user_opts => { with_events => 1 },
+                    room_opts => { room_version => "6" },
+                   ),
+                 federation_user_id_fixture() ],
+
+   do => sub {
+      my ( $outbound_client, $inbound_server, $creator, $room_id, $user_id ) = @_;
+      my $first_home_server = $creator->server_name;
+
+      my $datastore         = $inbound_server->datastore;
+
+      $outbound_client->join_room(
+         server_name => $first_home_server,
+         room_id     => $room_id,
+         user_id     => $user_id,
+         version     => "6",
+      )->then( sub {
```

Comment:
it's generally easier to use `federated_rooms_fixture` than separate calls to `local_user_and_room_fixtures`, `federation_user_id_fixture` and `join_room`.

Reply:
I think I caught the potential places this could be replaced...